### PR TITLE
spack recipes for AOCL-3.0 release

### PR DESCRIPTION
**Pull request by amd-toolchain-support**

### CODE REVIEW

File: var/spack/repos/builtin/packages/aocl-sparse/package.py [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+
+        if self.spec.variants['build_type'].value == 'Debug':
+            builddir = join_path(self.stage.source_path, 'build', 'debug')
+        else:
+            builddir = join_path(self.stage.source_path, 'build', 'release')
+
+        mkdirp(builddir)
+        return builddir
+
+    def cmake(self, spec, prefix):
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/aocl-sparse/package.py:55]: Why not implement `cmake_args` to create and return the list of arguments and leave lines 86-87 to the base class?

File: var/spack/repos/builtin/packages/aocl-sparse/package.py [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+
+    variant('build_type', default='Release',
+            description='CMake build type',
+            values=('Debug', 'Release'))
+    variant('shared', default=True,
+            description='Build shared library')
+    variant('ilp64', default=False,
+            description='Build with ILP64 support')
+
+    depends_on('boost', when='@2.2')
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/aocl-sparse/package.py:36]: Is `boost` really a dependency only for version 2.2?

File: var/spack/repos/builtin/packages/amdblis/package.py [hunk 3]
```diff
@@ -17,9 +17,44 @@ class Amdblis(BlisBase):
...
-    url = "https://github.com/amd/blis/archive/2.2.tar.gz"
+    url = "https://github.com/amd/blis/archive/3.0.tar.gz"
     git = "https://github.com/amd/blis.git"
 
     maintainers = ['amd-toolchain-support']
 
+    version('3.0', sha256='19ec56cb1c5855a2de22914acfa22fa4e23222992b392c33a2d73a9a8b516667')
     version('2.2', sha256='e1feb60ac919cf6d233c43c424f6a8a11eab2c62c2c6e3f2652c15ee9063c0c9')
+
+    def configure(self, spec, prefix):
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/amdblis/package.py:28]: There's a lot of duplication here with `BlisBase`'s `configure`.  Why not add an `configure_args` method here and in `BlisBase` to eliminate the duplication?

**Comment by tldahlgren** [comment]
Closing and re-opening to force re-running tests since there is a strange gateway error from the clingo tests:

```
Err:50 https://dl.bintray.com/sbt/debian  Release 
63  502  Bad Gateway [IP: 52.35.223.108 443]
64  Reading package lists...
65  E: The repository 'https://dl.bintray.com/sbt/debian  Release' no longer has a Release file.
```

**Comment by tldahlgren** [comment]
> Closing and re-opening to force re-running tests since there is a strange gateway error from the clingo tests:
> 
> ```
> Err:50 https://dl.bintray.com/sbt/debian  Release 
> 63  502  Bad Gateway [IP: 52.35.223.108 443]
> 64  Reading package lists...
> 65  E: The repository 'https://dl.bintray.com/sbt/debian  Release' no longer has a Release file.
> ```

This is a known issue -- https://github.com/actions/runner/issues/932 -- will try to close and re-open again.

**Review by tldahlgren — CHANGES_REQUESTED** [review]
Some of the failures being reported *appear* to be intermittent.  

Please resolve the conflicts for `aocl-parse` caused by merging #22079 .  Hopefully either they'll resolve the runner action issue or we'll get lucky on the next round.

**Comment by amd-toolchain-support** [comment]
Closing this pull request has rebase of aocl-sparse resulting in below error:

Your branch and 'origin/aocl_3_0_0' have diverged,
and have 56 and 3 different commits each, respectively.

Will raise another pull request and give this reference.

**Comment by amd-toolchain-support** [comment]
Closing this pull request has rebase of aocl-sparse resulting in below error:

Your branch and 'origin/aocl_3_0_0' have diverged,
and have 56 and 3 different commits each, respectively.

Will raise another pull request and give this reference.

**Comment by amd-toolchain-support** [comment]
New pull request:
https://github.com/spack/spack/pull/22376

**Pull request closed without merging**